lib/ansible/modules/network/cnos/cnos_vlan.py

```diff
@@ -223,66 +215,39 @@ def main():
             vlanArg5=dict(required=False),),
         supports_check_mode=False)
 
-    username = module.params['username']
-    password = module.params['password']
-    enablePassword = module.params['enablePassword']
-    vlanArg1 = module.params['vlanArg1']
-    vlanArg2 = module.params['vlanArg2']
-    vlanArg3 = module.params['vlanArg3']
-    vlanArg4 = module.params['vlanArg4']
-    vlanArg5 = module.params['vlanArg5']
     outputfile = module.params['outputfile']
-    hostIP = module.params['host']
-    deviceType = module.params['deviceType']
 
     output = ""
-    if not HAS_PARAMIKO:
-        module.fail_json(msg='paramiko is required for this module')
-
-    # Create instance of SSHClient object
-    remote_conn_pre = paramiko.SSHClient()
-
-    # Automatically add untrusted hosts (make sure okay for security policy in
-    # your environment)
-    remote_conn_pre.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-
-    # initiate SSH connection with the switch
-    remote_conn_pre.connect(hostIP, username=username, password=password)
-    time.sleep(2)
-
-    # Use invoke_shell to establish an 'interactive session'
-    remote_conn = remote_conn_pre.invoke_shell()
     time.sleep(2)
 
     # Enable and enter configure terminal then send command
-    output = output + cnos.waitForDeviceResponse("\n", ">", 2, remote_conn)
-
-    output = output + \
-        cnos.enterEnableModeForDevice(enablePassword, 3, remote_conn)
+    cmd = [{'command': '\n', 'prompt': None, 'answer': None}]
+    output = output + str(cnos.run_cnos_commands(module, cmd))
 
     # Make terminal length = 0
-    output = output + \
-        cnos.waitForDeviceResponse("terminal length 0\n", "#", 2, remote_conn)
+    cmd = [{'command': 'terminal length 0', 'prompt': None, 'answer': None}]
+    output = output + str(cnos.run_cnos_commands(module, cmd))
 
     # Go to config mode
-    output = output + \
-        cnos.waitForDeviceResponse("configure device\n", "(config)#", 2, remote_conn)
+    cmd = [{'command': 'configure terminal', 'prompt': None, 'answer': None}]
+    output = output + str(cnos.run_cnos_commands(module, cmd))
+    if(output.find('Error') != -1):
+        cmd = [{'command': 'configure device', 'prompt': None, 'answer': None}]
```

Comment:
if you use `edit_config` instead of `run_commands` in `vlanConfig`, you can avoid having to specify `configure device` in the module.

This would mean losing the customization added with `run_cnos_commands`, but I'm not totally clear what the point of all that is anyway. You would probably want to modify `edit_config` in the cliconf plugin to return responses, but that's fairly straightforward.

Reply:
I will check this method

Reply:
How to call this edit_config method in cliconf plugin in my module_util/cnos.py ? Do you have any sample to have a look. Greping the code, I could find some code of iosxr using it from Connection object. Please help me here.

Reply:
I prefer not to remove the above snippet. Why because: For CNOS 10.7.x onwards  CLIs are changing. Instead of "configure device", "configure terminal"  will work and instead of display show command will work and there are a lot of CLIs are changing. But I have to ensure backward compatibility for our modules. So what I am doing is to attempt with new CLI first, if thrown exception from device, attempt the older CLI . That's why the above logic is in place. Then I will test on both cnos version devices. I need to modify the cliconf/cnos.py also accordingly. 
At the moment i dont know how to call a cliconf/cnos.py method from my util cnos.py. Do u have any examples where people have done it?

Reply:
Additionally, for `configure terminal` vs `configure device`, cliconf has a `get_device_info()` method which, among other things, will report the current device version so that you can make decisions about what commands to use, instead of guessing and catching exceptions.

Reply:
Hi
Pardon me for my limitations in knowledge here
I tried to use the snippet like this to create a vlan 

def createVlan( module, prompt, answer):

    # vlan config command happens here. It creates if not present
    #command = "vlan " + vlanArg1 + "\n"
    timeout = 2
    vlanArg1 = module.params['vlanArg1']
    vlanArg2 = module.params['vlanArg2']
    vlanArg3 = module.params['vlanArg3']
    vlanArg4 = module.params['vlanArg4']
    vlanArg5 = module.params['vlanArg5']
    deviceType = module.params['deviceType']

    command = 'vlan ' + vlanArg1
    conn = get_connection(module)
    conn.get('enable')
    conn.get('configure terminal')
    retVal = conn.edit_config('vlan 13')

This code is not going to edit_config in cliconf/cnos.py
And I dont add conn.get('configure terminal') it will crib on privilege escalation issue.
Can you help me out here. How can I call the edit_config method in cliconf/cnos.py
I am struggling on this for past 3 days. 
Is it not better to wrap around my CLI  with the 'configure terminal' and 'end'  Or else I can define another method edit_config i my module_util.

Please help me